[tagger] Expose a gRPC tagger service in the DCA

juliogreff opened the pull request:
### What does this PR do?

This is a small refactoring of the gRPC tagger server in the core agent, to be able to also expose it in the DCA. Future clients of this service will be the remote taggers running on the cluster checks runners, in order to be able to take advantage of cluster level tags.

### Describe how to test/QA your changes

I'm adding the `skip-qa` on this one, as the core agent tagger is touched by everything and basically QA'd by everything else, and the DCA tagger is currently empty, so there's nothing to QA there yet, upcoming PRs will take care of that.

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

juliogreff committed:
[tagger] Expose a gRPC tagger service in the DCA

This is a small refactoring of the gRPC tagger server in the core agent,
to be able to also expose it in the DCA. Future clients of this
service will be the remote taggers running on the cluster checks
runners, in order to be able to take advantage of cluster level tags.

juliogreff committed:
[tagger] Remove health checks

First of all, this health check has never worked, as in, it has never
consistently passed. We just never noticed because the other agents
(process, trace, ...) don't actually have health checks configured.

Second of all, there's no obvious way to make it work with a gRPC
stream, since it is designed to block until there's a new message.
Having this receive on a separate goroutine is pointless as well, since
it's the very thing we want to monitor for "stuck-ness". So I'm electing
to have this removed entirely.

truthbk commented on pkg/tagger/remote/tagger.go:
```diff
@@ -70,7 +68,6 @@ func NewTagger() *Tagger {
 // Init initializes the connection to the remote tagger and starts watching for
 // events.
 func (t *Tagger) Init(ctx context.Context) error {
-	t.health = health.RegisterLiveness("tagger")
```
Do we not want any health reporting anymore for the tagger?

juliogreff replied:
From the second commit in this PR:

> First of all, this health check has never worked, as in, it has never
consistently passed. We just never noticed because the other agents
(process, trace, ...) don't actually have health checks configured.
>
> Second of all, there's no obvious way to make it work with a gRPC
stream, since it is designed to block until there's a new message.
Having this receive on a separate goroutine is pointless as well, since
it's the very thing we want to monitor for "stuck-ness". So I'm electing
to have this removed entirely.

truthbk reviewed (commented):
This looks awesome; thank you for refactoring this so cleanly. I have a couple of nits and a question, but this look pretty much good to go.

juliogreff committed:
Apply code review suggestions

* Move server.Server type definition to top of the file
* Set http.Server.Addr directly in NewMuxedGRPCServer

juliogreff commented:
Hey @truthbk, I addressed your comments in [317e5f0](https://github.com/DataDog/datadog-agent/pull/13844/commits/317e5f02b5ce5638694c3b014ce229f5e152e1ff). Are we good to go?
